andela/ah-codeblooded-frontend
# #160577579 Enable Article Like and Dislike

## What does this PR do?
The PR enables authenticated user to appreciate articles by either liking or disliking them.

## Description of Task to be completed?

##### Why this is important
Lets appreciate those authors who go out of their way to create good content with a Like and encourage those authors who do not meet our expectations with a dislike.

##### As a User
I want to be able to like or dislike articles  articles, both authored by me and those not authored by me.

##### Acceptance Criteria
Scenario:
Only **authenticated** users should be able to like or dislike articles.


## How can it be manually tested?
Clone the repository to your local environment.
```
git clone https://github.com/andela/ah-codeblooded-frontend.git
```

Change directories
```
cd ah-codeblooded-frontend/
```

Install the dependencies
```
yarn install
```

Run the server
```
yarn start
```

### It will open a browser with the current route at `/` and view the list of articles.

### Backend setup

- Authenticate a user on using the `ah-code-blooded` backend 
- Add the user credentials in the browser local storage using this this format:
```
{
        "username" : "<username>",
        "bio" : "<bio>",
        "avatar" : "",
        "token" : <token>
}
```
### Test Like/Dislike
- View one of the pre-existing articles on the home page
- Scroll to the bottom of the article and you will see the "like" and "dislike" button icons as shown on the screenshot below

**Screenshots (if appropriate)**
![image](https://user-images.githubusercontent.com/31212441/48483786-7211c000-e825-11e8-9591-decd68afc0ee.png)


**What are the relevant pivotal tracker stories?**

The relevant pivotal story is [here](https://www.pivotaltracker.com/n/projects/2198535/stories/160577579)
